File: src/js/component/tray.ts
import {TrayIcon, TrayIconOptions} from '@tauri-apps/api/tray';
import {Menu} from '@tauri-apps/api/menu';
import {defaultWindowIcon} from '@tauri-apps/api/app';
import {getCurrentWindow} from "@tauri-apps/api/window";


export async function initTray(): Promise<void> {
    const oldTray = await TrayIcon.getById('main-tray');
    if (oldTray) {
        await oldTray.setMenu(null);
        await oldTray.close();
    }

    const env = await import('../render/env');
    const player = await import('../render/player');

    const menu = await Menu.new({
        id: 'tray-menu',
        items: [
            {
                id: 'forward',
                text: '上一曲',
                action: player.anonymous_fun.skipForward
            },
            {
                id: 'pause-play',
                text: '播放/暂停',
                action: () => env.pauseToggle()
            },
            {
                id: 'backward',
                text: '下一曲',
                action: player.anonymous_fun.skipBackward
            },
            {
                id: 'quit',
                text: '退出',
                action: () => getCurrentWindow().emit('quit'),
            },
        ],
    });

    const options: TrayIconOptions = {
        id: 'main-tray',
        menu: menu,
        title: 'Lumen Audio Player',
        icon: await defaultWindowIcon() ?? undefined,
        showMenuOnLeftClick: false,
        action: async (event) => {
            if (event.type === 'DoubleClick') {
                const window = getCurrentWindow();
                await window.show();
                await window.unminimize();
                await window.setFocus()
            }
        },
    }

    await TrayIcon.new(options);
}

File: src-tauri/src/tray.rs
use crate::window::show_window;
use tauri::menu::{Menu, MenuItem};
use tauri::tray::{TrayIconBuilder, TrayIconEvent};
use tauri::{AppHandle, Emitter, Manager};

pub fn create_tray(app: &AppHandle) -> Result<(), Box<dyn std::error::Error>> {
    let open_window = MenuItem::with_id(app, "forward", "上一曲", true, None::<&str>)?;
    let sync_data = MenuItem::with_id(app, "pause-play", "播放/暂停", true, None::<&str>)?;
    let show_settings = MenuItem::with_id(app, "backward", "下一曲", true, None::<&str>)?;
    let quit = MenuItem::with_id(app, "quit", "退出", true, None::<&str>)?;

    let menu = Menu::with_items(app, &[&open_window, &sync_data, &show_settings, &quit])?;

    TrayIconBuilder::new()
        .icon(app.default_window_icon().unwrap().clone())
        .tooltip("Lumen Audio Player")
        .show_menu_on_left_click(false)
        .menu(&menu)
        .on_tray_icon_event(|tray, event| match event {
            TrayIconEvent::DoubleClick { .. } => show_window(tray.app_handle()),
            _ => {}
        })
        .on_menu_event(|app, event| match event.id.as_ref() {
            "quit" => {
                if let Some(window) = app.get_webview_window("main") {
                    let _ = window.close();
                }
            }
            other => {
                app.emit("tray://action", other).ok();
            }
        })
        .build(app)?;

    Ok(())
}
